### ISSUE TITLE
yoloe-26 models need

### ISSUE THREAD

**Issue by jiuzhuanzhuan** [ultralytics/ultralytics#24001]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I’m looking for a YOLOE26 model without the segmentation head, or alternatively a YOLOE26-Seg model that does not use BatchNormal fusion.

Currently, I’m trying to fine-tune the model on my own domain, but I’m facing some issues:

Without pretrained weights, the performance is very poor.
For YOLOE26-Seg, the use of batch fusion also leads to unsatisfactory results during fine-tuning.

So I was wondering:

Is there a version of YOLOE26 without segmentation available?
Or is there a way to disable BatchNorm fusion in YOLOE26-Seg?

<img width="636" height="376" alt="Image" src="https://github.com/user-attachments/assets/12fa92fb-be7a-426b-94f5-faa528792e4c" />

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jiuzhuanzhuan, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue moving quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
Since this appears to be a model/training behavior question related to YOLOE26 fine-tuning, please provide as much information as possible, including:
- the exact model file or checkpoint you are using
- your training command or script
- your `ultralytics` version
- your environment details
- dataset image examples
- training logs
- whether the issue reproduces after upgrading
If this is also related to unexpected model behavior or a possible 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
The training shouldn't use fused model. Fused model is only used for inference.

**Comment by jiuzhuanzhuan** [comment]
> The training shouldn't use fused model. Fused model is only used for inference.

But open-source models are only available in the fused version.
https://docs.ultralytics.com/zh/models/yolo26/#yoloe-26-open-vocabulary-instance-segmentation

**Comment by jiuzhuanzhuan** [comment]
> The training shouldn't use fused model. Fused model is only used for inference.

The problem I’m facing is that training performs poorly if I don’t warm up with your pre-trained model. So I need to load an unfused model for warm-up. I’m currently trying to set the BN mean to 0 and variance to 1, then re-estimate the parameters so that the model with BN produces inference results close to the fused model, but it hasn’t worked yet. Another option I’m considering is initializing training from YOLO-26s.

**Comment by Y-T-G** [comment]
Did you try fine-tuning using this?

https://docs.ultralytics.com/models/yoloe/#fine-tuning

**Comment by jiuzhuanzhuan** [comment]
> Did you try fine-tuning using this?
> 
> https://docs.ultralytics.com/models/yoloe/#fine-tuning

Yes, I used the following code for fine-tuning:

```from ultralytics import YOLOE
from ultralytics.models.yolo.yoloe import YOLOEPETrainer

# Initialize a detection model from a config
model = YOLOE("yoloe-26s.yaml")

# Load weights from a pretrained segmentation checkpoint (same scale)
model.load("yoloe-26s-seg.pt")

# Fine-tune on your detection dataset
results = model.train(
    data="coco128.yaml",  # Detection dataset
    epochs=80,
    patience=10,
    trainer=YOLOEPETrainer,  # <- Important: use detection trainer
)
```


The only issue is that yoloe-26s-seg.pt has already been fused. As I showed before ([link](https://github.com/ultralytics/ultralytics/issues/24001#issue-4132048014)
), fine-tuning with warm-up from a non-fused model yields better results.

<img width="1139" height="457" alt="Image" src="https://github.com/user-attachments/assets/95026e11-ce56-4123-aefe-ef379360422d" />

**Comment by Y-T-G** [comment]
@ShuaiLYU Is non-fused YOLOE26 checkpoints available?

**Comment by ShuaiLYU** [comment]
> [@ShuaiLYU](https://github.com/ShuaiLYU) Is non-fused YOLOE26 checkpoints available?

Let’s talk about this internally first. @Laughing-q

**Comment by ShuaiLYU** [comment]
@jiuzhuanzhuan   please find the unfused yoloe-26s-seg.pt  (batchnorm unfused) . Any problem please come to me. https://drive.google.com/file/d/1TrNV4tAxEYRF-tL_MUh2NaOpSItUU2c9/view?usp=drive_link

**Comment by jiuzhuanzhuan** [comment]
> [@jiuzhuanzhuan](https://github.com/jiuzhuanzhuan) please find the unfused yoloe-26s-seg.pt (batchnorm unfused) . Any problem please come to me. https://drive.google.com/file/d/1TrNV4tAxEYRF-tL_MUh2NaOpSItUU2c9/view?usp=drive_link

Thank you very much, that worked!
<img width="626" height="537" alt="Image" src="https://github.com/user-attachments/assets/94460bc3-982a-4d29-9fdc-01ff92a6f7ef" />

**Comment by glenn-jocher** [comment]
Glad to hear the unfused checkpoint resolved it; thanks for confirming, and if anything else comes up with `YOLOE` fine-tuning, please open a new issue with a small reproducible example so the team can track it clearly.

**Comment by yang-0201** [comment]
I found that when fine-tuning with different checkpoints, the printed weight-loading information can reflect this issue to some extent. For example, when using

model = YOLOE("yoloe-26s-seg(bn_unfused).pt")

it prints:

Transferred 705/707 items from pretrained weight

but when using

model = YOLOE("yoloe-26s-seg.pt")

it prints:

Transferred 155/707 items from pretrained weight

Would it be possible to release all of the unfused-BN checkpoints in the future?

**Comment by ShuaiLYU** [comment]
> I found that when fine-tuning with different checkpoints, the printed weight-loading information can reflect this issue to some extent. For example, when using
> 
> model = YOLOE("yoloe-26s-seg(bn_unfused).pt")
> 
> it prints:
> 
> Transferred 705/707 items from pretrained weight
> 
> but when using
> 
> model = YOLOE("yoloe-26s-seg.pt")
> 
> it prints:
> 
> Transferred 155/707 items from pretrained weight
> 
> Would it be possible to release all of the unfused-BN checkpoints in the future?

Yep, we are going to release them ASAP.
